Documentation Extension (#5)
eprei/ft_transcendence: docs/README.adoc

= Transcendence

> :nofooter:

Reviewer: Resource: [active the table of contents](https://docs.asciidoctor.org/asciidoc/latest/toc/#activate-the-toc)
```suggestion
:nofooter:
:toc: left
```

Reviewer 2: that's a good idea

Author: Stylé

portin_postgre -up-> portout_postgre
NestJS <-le-> PostgreSQL
....

== Project overview

=== Page 1: Welcome

==== Functionalities:

* Login with the intranet OAuth system 42

==== Technical requirements:

* Communication with OAuth 42

==== Graphical requirements:

* Background, login button

image::assets/Page1.1_Welcome.png[welcome preview]

=== Page 2: Compte utilisateur

==== 2.1 Profile

==== Functionalities:

* You can choose a (unique) username to be displayed on the site. We start with the default name taken from OAuth 42 (?)

> * You can upload an avatar. If you don't, it defaults to your 42 profile picture (?)

Reviewer: better ?
```suggestion
* You can upload an avatar. If you don't, it is by default to your 42 profile picture (?)
```

* Join a channel (not requested by the subject but it is logical to add it)
* Send direct messages to other users
* Block other users
* Invite other users to play pong
* Access to other players' profiles

==== Technical requirements:

_To be completed..._

==== Graphic requirements:

_Design made directly from the description of the subject to optimise time_

image::assets/Page3_Chat.png[chat preview]

=== Page 4 - The game

==== Functionalities:

* Launch the game of pong against another player
* Matching system (waiting list)
* Canvas with the game
* Customisation options (we have to define which ones)
* The game must be responsive

==== Technical needs:

_To be completed..._

==== Graphic needs:

_Design made directly from the description of the subject to optimize time_

image::assets/Page4.1_GameLauncher.png[game preview]
image::assets/Page4.2_TheGame.png[game preview]

=== Navbar

. Logo (Just the 'CosmicPong' word)
. Profil
. Match History
. Chat
. The game
. LogOut (?)
. Logo 42Lausanne

image::assets/Navbar.png[Navbar preview]

=== Window scheme

[plantuml, target=assets/design, format=svg, width=100%]
....
rectangle {

	object Page01
	object Page02
	object Page03

> 	object Page04

Author: Nice method

> 	object Page04

Reviewer: > What is the most efficient way to proceed now?

IMHO, the most efficient is to do a commit per comment
and to reply to the comment with the commit ID,
this way, the commenter can check if the commit matches the expectations.

	object Welcome {
		Login with the intranet OAuth system 42
		Communication with OAuth 42
	}

	object Profile {
		Choose username
		Choose/upload an avatar
		Enable 2FA

> 		Add users ass friends and view real time status

Reviewer: typo ?
```suggestion
		Add users as friends and view real time status
```
Split this sentence:
```suggestion
		Add users as friends
		See the status of friends in real time
```

Author: Done

_To be completed..._

==== Graphic needs:

_Design made directly from the description of the subject to optimize time_

image::assets/Page4.1_GameLauncher.png[game preview]
image::assets/Page4.2_TheGame.png[game preview]

=== Navigation bar

. Logo (Just the 'CosmicPong' word)
. Profil
. Match History
. Chat
. The game
. LogOut (?)
. Logo 42Lausanne

image::assets/Navbar.png[Navbar preview]

=== Window scheme

[plantuml, target=assets/design, format=svg, width=100%]
....
rectangle {

	object "Page 1" as Page1
	object "Page 2" as Page2
	object "Page 3" as Page3
	object "Page 4" as Page4

	object Welcome {
		Login with the intranet OAuth system 42
		Communication with OAuth 42
	}

	object Profile {
		Choose username
		Choose/upload an avatar
		Enable 2FA
		Add users as friends
		See the status of friends in real time
	}

	object MatchHistory{
		Display of history including 1 vs 1
	}

	object Chat{
		Create channels
		Join channels
		Send direct messages to other users
		Block other users
		Invite other users to play pong
		Access to other players' profiles
	}

	object Navbar{
		CosmicPong (logo)
		Profil
		Match History
		Chat
		The game
		LogOut (?)
		42Lausanne (logo)
	}

> 	object TheGame{

Reviewer: - Adding space ?
- Add a space before the curly brace `{` ?
```suggestion
	object "The game" as TheGame {
```
